From cc834c8e0b4daefa75f6900daf08846780b1780c Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 19 May 2017 17:33:32 +0200
Subject: [PATCH 0700/1200] net: mvpp2: Allow to support 802.3ap Backplane
 Auto-Negotiation

When this mode is chosen all mac/pcs/serdes related configuration should be
skipped, since the dedicated co-processor (MG's CM3) runs auto-negotiation
firmware and configures all relevant registers.

The co-processor has access to all network related registers like: MG, AP,
MAC, PCS, Serdes etc. and during auto-neg process it accesses them.
Therefore the mvpp2.c driver should not interfere and stay away from
accessing mac related registers.

Change-Id: I38e97454a9e1da8bc69ed4df53a23b81f2dc231e
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/55397
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: Igal Liberman <igall@marvell.com>
---
 drivers/net/mvpp2.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/mvpp2.c b/drivers/net/mvpp2.c
index 8891d9b664..d903568e3d 100644
--- a/drivers/net/mvpp2.c
+++ b/drivers/net/mvpp2.c
@@ -3399,6 +3399,9 @@ static int gop_port_init(struct mvpp2_port *port)
 		gop_xlg_mac_reset(port, 0);
 		break;
 
+	case PHY_INTERFACE_MODE_AP:
+		break;
+
 	default:
 		netdev_err(NULL, "%s: Requested port mode (%d) not supported\n",
 			   __func__, port->phy_interface);
@@ -3439,6 +3442,8 @@ static void gop_port_enable(struct mvpp2_port *port, int enable)
 	case PHY_INTERFACE_MODE_SFI:
 		gop_xlg_mac_port_enable(port, enable);
 
+		break;
+	case PHY_INTERFACE_MODE_AP:
 		break;
 	default:
 		netdev_err(NULL, "%s: Wrong port mode (%d)\n", __func__,
-- 
2.22.0

